ksql-engine/src/main/java/io/confluent/ksql/planner/plan/KsqlStructuredDataOutputNode.java

```diff
@@ -292,10 +293,12 @@ private static SourceTopicProperties getSinkTopicPropertiesLegacyWay(
   ) {
     final int partitions = (Integer) sinkProperties.getOrDefault(
         KsqlConfig.SINK_NUMBER_OF_PARTITIONS_PROPERTY,
-        ksqlConfig.getInt(KsqlConfig.SINK_NUMBER_OF_PARTITIONS_PROPERTY));
+        Optional.ofNullable(ksqlConfig.getInt(KsqlConfig.SINK_NUMBER_OF_PARTITIONS_PROPERTY))
+            .orElse(KsqlConstants.legacyDefaultSinkPartitionCount));
```

Comment:
Don't we have this in the `KSQLConfig` for the legacy case? In the legacy case `KSQLConfig` will provide `KsqlConstants.legacyDefaultSinkPartitionCount`.

Reply:
The problem is that it is possible for a user to set `ksql.sink.replicas` and/or `ksql.sink.partitions` in non-legacy mode. When this happens, the code assumes the user is in legacy mode, but if they've only set one of the two properties, the other takes the current default value of null, resulting in the null pointer exception.